### PR TITLE
post method support

### PR DESCRIPTION
**Pull request by AkashSinghPanwar-Dassana**

### CODE REVIEW

**Comment by AkashSinghPanwar-Dassana** [comment]
in ``statusValidator`` method what error should be raised ?

**Comment by akashdeep-dassana** [comment]
Can we have a timeout for the request if the server doesn't respond in time? 

https://requests.readthedocs.io/en/latest/user/quickstart/#timeouts

File: dassana/common.py [hunk 1]
```diff
@@ -8,7 +8,9 @@
 import logging
 from .dassana_env import *
 import datetime
-from tenacity import retry, wait_fixed, stop_after_attempt, before_sleep_log
+from tenacity import retry, wait_fixed, stop_after_attempt, before_sleep_log, retry_if_exception_type
+from werkzeug.exceptions import InternalServerError, BadRequest, NotFound, Unauthorized, RequestTimeout
+from requests.exceptions import ReadTimeout
```
Comment (akashdeep-dassana) [inline dassana/common.py:13]: Use `Timeout` it handles both `ConnectTimeout` and `ReadTimeout`

**Comment by AkashSinghPanwar-Dassana** [comment]
need to discuss about socket timeout exception

**Pull request closed without merging**